app/graphql/mutations/create_group.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Mutations
+  # Base Mutation
+  class CreateGroup < BaseMutation
+    null true
+    description 'Create a new group..'
+    argument :description, String, required: false, description: 'The description to give the group.'
+    argument :group_id, ID,
+             required: false,
+             description: 'The Node ID of a group to create the sub group in. For example, `gid://irida/Group/a84cd757-dedb-4c64-8b01-097020163077`.' # rubocop:disable Layout/LineLength
+    argument :group_puid, ID, # rubocop:disable GraphQL/ExtractInputType
+             required: false,
+             description: 'Persistent Unique Identifier of a group to create the sub group in. For example, `INXT_GRP_AAAAAAAAAA`.' # rubocop:disable Layout/LineLength
+    argument :name, String, required: true, description: 'The name to give the group.' # rubocop:disable GraphQL/ExtractInputType
+    argument :path, String, required: false, description: 'A custom path for the group.' # rubocop:disable GraphQL/ExtractInputType
+
+    field :errors, [Types::UserErrorType], null: false, description: 'A list of errors that prevented the mutation.'
+    field :group, Types::GroupType, description: 'The newly created group.'
+
+    def resolve(args)
+      parent_group = nil
+
+      if args[:group_id] || args[:group_puid]
+        parent_group = get_group_from_id_or_puid_args(args)
+
+        if parent_group.nil? || !parent_group.persisted?
+          user_errors = [{ path: ['group'], message: 'Group not found by provided ID or PUID' }]
+          return { group: nil, errors: user_errors }
+        end
+      end
+
+      # slugify path, if no path given slugify name to use as path
+      args[:path] = if args[:path]
+                      args[:path].to_s.parameterize(separator: '-')
```

Comment:
**nitpick (blocking):** Can you only slugify path only if you have to generate it from name (e.g. path is not sent as a argument). Otherwise accept the path provided and allow that to raise an error if it does not pass validation.

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/799/commits/e4202e61703f587b14214c1d94d38a032757fbc3